stix2/sdo.py

```diff
@@ -6,14 +6,15 @@
 
 from .base import _STIXBase
 from .common import ExternalReference, GranularMarking, KillChainPhase
+from .markings import MarkingsMixin
 from .observables import ObservableProperty
 from .properties import (BooleanProperty, IDProperty, IntegerProperty,
                          ListProperty, PatternProperty, ReferenceProperty,
                          StringProperty, TimestampProperty, TypeProperty)
 from .utils import NOW
 
 
-class AttackPattern(_STIXBase):
+class AttackPattern(_STIXBase, MarkingsMixin):
```

Comment:
I think it's time to make an `STIXDomainObject` class that subclasses `_STIXBase` and `MarkingsMixin` and use that as the superclass for all SDOs.